server/src/Types.hs

```diff
@@ -77,14 +80,42 @@ data ServerOptions = ServerOptions
   { port :: Port
   , nodeSocket :: FilePath
   , networkId :: C.NetworkId
+  , ogmiosHost :: String
+  , ogmiosPort :: Int
   }
   deriving stock (Generic)
 
 newEnvIO :: ServerOptions -> IO (Either String Env)
-newEnvIO serverOptions =
-  getDataFileName "config/pparams.json"
-    >>= Aeson.eitherDecodeFileStrict @Shelley.ProtocolParameters
-    <&> second (Env serverOptions)
+newEnvIO serverOptions@ServerOptions {..} =
+  let ogmiosServerParams =
+        Ogmios.Query.defaultServerParameters
+          { Ogmios.Query.getPort = ogmiosPort
+          , Ogmios.Query.getHost = ogmiosHost
+          }
+      queryProtocolParams :: IO LC8.ByteString
+      queryProtocolParams =
+        Ogmios.Query.makeRequest
+          ogmiosServerParams
+          $ Ogmios.Query.Query
+            Ogmios.Query.CurrentProtocolParameters
+      maxConnectionAttempts = 300
+   in do
+        eitherResponse <-
+          Ogmios.Query.tryQueryUntilZero
+            queryProtocolParams
+            maxConnectionAttempts
+        case eitherResponse of
+          Right response ->
+            case decodeProtocolParameters response of
+              Right params ->
+                (return . Right . Env serverOptions) params
```

Comment:
It looks strange to me to use function composition and then use parens for application (instead of `$`)